web/web/static/js/components/top_bar.js

```diff
@@ -1,26 +1,45 @@
-import '@rmwc/top-app-bar/styles';
-
-import React from 'react';
-import { connect } from 'react-redux';
-import { SimpleTopAppBar, TopAppBarFixedAdjust } from '@rmwc/top-app-bar';
-import { toggleNavigationDrawer, logout } from '../actions';
+import "@rmwc/top-app-bar/styles";
 
+import React from "react";
+import { connect } from "react-redux";
+import { SimpleTopAppBar, TopAppBarFixedAdjust } from "@rmwc/top-app-bar";
+import { toggleNavigationDrawer, completeLogout, resetUserLoggedOut } from "../actions";
+import { createErrorMessage } from "../helpers";
+import { api } from "../network_client";
 
 class TopBar extends React.Component {
     toggleNavigationDrawer = () => {
         this.props.dispatch(toggleNavigationDrawer());
-    }
+    };
+
+    logout = () => {
+        try {
+            api.delete(
+                "logout",
+                { json: { logout: true } },
+                () => {
+                    this.props.dispatch(completeLogout());
+                },
+                () => {
+                    createErrorMessage("Error", "Unable to logout");
```

Comment:
I feel like failing a logout operation is weird. I think even if the backend is unable to clear the access token, we can and should still clear everything on the web client (localStorage) and redirect the user to the login page. Does doing this cause any other issues?

Reply:
I don't think this would cause other issues. Will fix.

Reply:
done